src/cdk/table/table.ts

```diff
@@ -232,19 +264,25 @@ export class CdkTable<T> implements CollectionViewer, OnInit, AfterContentChecke
     if (this._headerRowDef) {
       this._headerRowDefChanged = true;
     }
+
+    // If the table has a footer row definition defined as part of its content, flag this as a
+    // header row def change so that the content check will render the header row.
+    if (this._footerRowDef) {
+      this._footerRowDefChanged = true;
+    }
   }
 
   ngAfterContentChecked() {
     // Cache the row and column definitions gathered by ContentChildren and programmatic injection.
     this._cacheRowDefs();
     this._cacheColumnDefs();
 
-    // Make sure that the user has at least added a header row or row def.
-    if (!this._headerRowDef && !this._rowDefs.length) {
+    // Make sure that the user has at least added header, footer, or data row def.
+    if (!this._headerRowDef && !this._footerRowDef && !this._rowDefs.length) {
```

Comment:
Currently the only requirement is that the table must have defined either a`mat-header-row`, `mat-row`. If you do not have either, then an error is thrown. If you do not have one of them, they just are not rendered.

This continues the check - if you do not have _any_ rows defined, an error is thrown. Otherwise, anything excluded just is not rendered.

Reply:
So you would say it's valid to have a footer row template defined with no header and no data row templates?

Reply:
Yes - mostly because this is necessary for some users with the header row and I'd prefer symmetry.

Some context: https://github.com/angular/material2/issues/7604